### PR TITLE
add HWM.update method

### PR DESCRIPTION
**Pull request by maxim-lixakov**
## Change Summary

- Add ``HWM.update`` method due to some implementation-specific logic in each hwm class 

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [ ] Unit and integration tests for the changes exist
* [ ] Tests pass on CI and coverage does not decrease
* [x] Documentation reflects the changes where applicable
* [x] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/etl-entities/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: etl_entities/hwm/hwm.py [hunk 1]
```diff
@@ -172,6 +172,10 @@ def deserialize(cls, inp: dict):
 
         return super().deserialize(value)
 
+    @abstractmethod
+    def update(self, value):
```
Comment (dolfinus) [inline etl_entities/hwm/hwm.py:176]:
```suggestion
    def update(self, value: ValueType):
```

File: etl_entities/hwm/column/column_hwm.py [hunk 2]
```diff
@@ -188,6 +188,43 @@ def __eq__(self, other):
 
         return self.value == other
 
+    def update(self, value: ColumnValueType):
```
Comment (dolfinus) [inline etl_entities/hwm/column/column_hwm.py:191]: Нужно добавить тесты на метод, можно взять из v1

File: etl_entities/hwm/column/column_hwm.py [hunk 3]
```diff
@@ -188,6 +188,43 @@ def __eq__(self, other):
...
+
+        .. note::
+
+            Changes HWM value in place
+
+        Returns
+        -------
+        result : ColumnHWM
+
+            HWM copy with new value
```
Comment (dolfinus) [inline etl_entities/hwm/column/column_hwm.py:202]:
```suggestion
            HWM with new value
```

File: etl_entities/hwm/hwm.py (hunk 1 above)
Comment (maxim-lixakov) [reply etl_entities/hwm/hwm.py:176]: в `FileListHWM`  - `def update(self, value: str | os.PathLike | Iterable[str | os.PathLike]):`, mypy ругается, вероятно по этой причине в `old_hwm.HWM` без аннотации указан `value`

Comment (dolfinus) [reply etl_entities/hwm/hwm.py:176]: Хм. Ну ок, пусть будет так

File: tests/test_hwm/test_column_hwm.py [hunk 4]
```diff
@@ -357,3 +357,74 @@ class UnregisteredHWM(hwm_class):
...
+    hwm4 = old_hwm4.update(value)
+
+    assert hwm4 == hwm3 == hwm1
+    assert hwm4.value == old_hwm4.value == hwm3.value == old_hwm3.value == value  # old object is returned
+    assert hwm3 is old_hwm3  # in-place replacement
+    assert hwm4 is old_hwm4  # in-place replacement
+    assert hwm4.modified_time == hwm3.modified_time == hwm1.modified_time
+
+    # if current value is less than input, use input as a new value and update modified_time
+    hwm5 = hwm1.copy().copy(update={"value": value + delta})
```
Comment (dolfinus) [inline tests/test_hwm/test_column_hwm.py:410]:
```suggestion
    hwm5 = hwm1.copy(update={"value": value + delta})
```

**Review by dolfinus — APPROVED** [review]

**Pull request merged**